pkg/ovs/openflow/ofctrl_action_test.go

```diff
@@ -0,0 +1,1476 @@
+// Copyright 2022 Antrea Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package openflow
+
+import (
+	"encoding/binary"
+	"fmt"
+	"net"
+	"testing"
+
+	"antrea.io/libOpenflow/openflow15"
+	"antrea.io/libOpenflow/util"
+	"antrea.io/ofnet/ofctrl"
+	"github.com/stretchr/testify/assert"
+)
+
+var (
+	tableID1   = uint8(1)
+	tableID2   = uint8(2)
+	tableName  = "testTable"
+	stageID    = StageID(0)
+	piplineID  = PipelineID(0)
+	missAction = TableMissActionNext
+)
+
+type actionSetField struct {
+	class uint16
+	field uint8
+
+	ethSrc        net.HardwareAddr
+	ethDst        net.HardwareAddr
+	arpXHa        net.HardwareAddr
+	arpXPa        net.IP
+	ipv4Src       net.IP
+	ipv4Dst       net.IP
+	ipv6Src       net.IP
+	ipv6Dst       net.IP
+	tunnelIPv4Dst net.IP
+	vlanId        uint16
+	arpOper       uint16
+
+	fieldValue util.Message
+	fieldMask  util.Message
+}
+type actionCopyField struct {
+	oxmIdSrcClass uint16
+	oxmIdDstClass uint16
+	oxmIdSrcField uint8
+	oxmIdDstField uint8
+	nBits         uint16
+	srcOffset     uint16
+	dstOffset     uint16
+}
+type nxActionOutputReg struct {
+	class    uint16
+	field    uint8
+	ofsNbits uint16
+	regID    uint8
+}
+type nxActionConnTrack struct {
+	flags        uint16
+	zoneSrc      uint32
+	zoneOfsNbits uint16
+	recircTable  uint8
+}
+type nxActionCTNAT struct {
+	flags         uint16
+	rangeIPv4Min  net.IP
+	rangeIPv4Max  net.IP
+	rangeIPv6Min  net.IP
+	rangeIPv6Max  net.IP
+	rangeProtoMin *uint16
+	rangeProtoMax *uint16
+}
+
+func newExpectedNXActionOutputReg(class uint16, field uint8, regID uint8, rng *Range) *nxActionOutputReg {
+	return &nxActionOutputReg{
+		class:    class,
+		field:    field,
+		ofsNbits: rng.ToNXRange().ToOfsBits(),
+		regID:    regID,
+	}
+}
+
+func checkNXActionOutputReg(t *testing.T, expected *nxActionOutputReg, action openflow15.Action) {
```

Comment:
I wonder why it creates a new struct nxActionOutputReg instead of just use `openflow15.NXActionOutputReg` as the expected result, with which it could just use assert.Equal to check result.

Reply:
At the beginning, I thought of that we just need verify some fields of struct like `openflow15.NXActionOutputReg`. Another reason is that if we use `openflow15.NXActionOutputReg`, it is nothing different in the code of building expected struct from the that in lib openflow.